graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/tracing/FederatedTracingInstrumentation.java

```diff
@@ -433,9 +429,27 @@ public Node(Reports.Trace.Node.Builder builder) {
 
     public static class Options {
         private final boolean debuggingEnabled;
+        private final Predicate<Object> contextSignalsToTrace;
```

Comment:
Wdyt about passing in the `ExecutionInput` rather than the `context`? Would implementors ever want to consult something besides the context - possibly the `GraphQLContext`? 

It would also have the benefit of the input being typed rather than `Object`.

Wdyt about naming it something more generic as well - like `shouldTrace`?

Reply:
Awesome suggestions!